tests/h/traversal/annotation_test.py

```diff
@@ -1,348 +1,134 @@
-from unittest import mock
+from unittest.mock import sentinel
 
-import pyramid.security
 import pytest
+from h_matchers import Any
 from pyramid import security
 from pyramid.authorization import ACLAuthorizationPolicy
 
 from h.security.permissions import Permission
 from h.traversal import AnnotationContext, AnnotationRoot
 
 
-class FakeGroup:
-    # NB: Tests that use this do not validate that the principals are correct
-    # for the indicated group. They validate that those principals are being
-    # transferred over to the annotation as expected
-    # As such, this has sort of a partial and wonky re-implementation of
-    # ``h.models.Group.__acl__``
-    # This is a symptom of the disease that is splitting ACL concerns between
-    # traversal/resources and model classes
-    # TODO: Refactor once we're able to move ACLs off of models
-    def __init__(self, principals):
-        acl = []
-        for p in principals:
-            acl.append((security.Allow, p, Permission.Group.READ))
-            if p == security.Everyone:
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.FLAG)
-                )
-                acl.append(
-                    (security.Allow, security.Authenticated, Permission.Group.MODERATE)
-                )
-            else:
-                acl.append((security.Allow, p, Permission.Group.FLAG))
-                # Normally, the ``moderate`` permission would only be applied
-                # to the admin (creator) of a group, but this ``FakeGroup``
-                # is indeed fake. Tests in this module are merely around whether
-                # this permission is translated appropriately from a group
-                # to an annotation context (i.e. it should not be applied
-                # to private annotations)
-                acl.append((security.Allow, p, Permission.Group.MODERATE))
-        self.__acl__ = acl
-
-
-@pytest.mark.usefixtures("groupfinder_service", "links_service")
 class TestAnnotationRoot:
-    def test_it_does_not_assign_create_permission_without_authenticated_user(
-        self, set_permissions, pyramid_request
-    ):
-        set_permissions()
-
-        context = AnnotationRoot(pyramid_request)
+    def test_create_permission_requires_authenticated_user(self, root, permits):
+        assert permits(root, [security.Authenticated], Permission.Annotation.CREATE)
+        assert not permits(root, [], Permission.Annotation.CREATE)
 
-        assert not pyramid_request.has_permission(Permission.Annotation.CREATE, context)
-
-    def test_it_assigns_create_permission_to_authenticated_request(
-        self, set_permissions, pyramid_request
+    def test_annotation_lookup(
```

Comment:
Not sure about the test name. Maybe something like `test_it_returns_the_right_AnnotationContext()`?